app/models/usage.rb

```diff
@@ -7,9 +7,18 @@ class Usage < ApplicationRecord
   scope :in_organization, ->(organization = Organization.current) { where(organization_id: organization.id) }
 
   before_save :set_slug
+  before_destroy :destroy_children_usages!
+
+  def self.destroy_all_where(query)
+    where(query).destroy_all
+  end
 
   def self.destroy_usages_for(record)
-    Usage.where(parent_item: record).destroy_all
+    destroy_all_where(parent_item: record)
+  end
+
+  def destroy_children_usages!
+    item.children.each { |child| Usage.destroy_all_where(item: child, organization: organization) }
```

Comment:
The performance improvement that we talked about isn't as simple as: 

`Usage.destroy_all_where(item: item.children, organization: organization)` ?

If so it would be nice to change it. If I'm missing something lets integrate it like this and leave a separate issue for that particular improvement.